lib/nexpose/wait.rb

```diff
@@ -0,0 +1,133 @@
+module Nexpose
+
+  class Wait
+    attr_reader :error_message, :ready, :retry_count, :timeout, :polling_interval
+
+
+    def initialize(retry_count: nil, timeout: nil, polling_interval: nil)
+      @error_message = "Default General Failure in Nexpose::Wait"
+      @ready = false
+      @retry_count = retry_count.nil? ? 0 : retry_count
+      @timeout = timeout
+      @polling_interval = polling_interval
+    end
+
+
+    def is_ready?
+      @ready
+    end
+
+
+    def for_report(nsc: nil, report_id: nil)
+      begin
+        poller = Nexpose::Poller.new(timeout: @timeout, polling_interval: @polling_interval)
+        poller.wait(get_report_status(nsc: nsc, report_id: report_id))
+        @ready = true
+      rescue TimeoutError
+        retry if timeout_retry?
+        @error_message = "Timeout Waiting for Report to Generate - Report Config ID: #{report_id}"
+      rescue NoMethodError => error
```

Comment:
Instead of rescuing `NoMethodError` let's use `#respond_to?` where the exception could be raised. See the following benchmark: https://github.com/JuanitoFatas/fast-ruby#readme
>`rescue` vs `respond_to?` for Control Flow code
>
>```
>$ ruby -v code/general/begin-rescue-vs-respond-to.rb
ruby 2.2.0p0 (2014-12-25 revision 49005) [x86_64-darwin14]
>
Calculating -------------------------------------
      begin...rescue    29.452k i/100ms
         respond_to?   106.528k i/100ms
-------------------------------------------------
      begin...rescue    371.591k (± 5.4%) i/s -      1.855M
         respond_to?      3.277M (± 7.5%) i/s -     16.299M
>
Comparison:
         respond_to?:  3276972.3 i/s
      begin...rescue:   371591.0 i/s - 8.82x slower
>```